syntax = "proto3";

service ScalogSequencer {
    // Notifies another broker to start their local sequencer for Scalog
    rpc HandleScalogStartLocalSequencer(ScalogStartLocalSequencerRequest) returns (ScalogStartLocalSequencerResponse);
    // Receives a local cut from a local sequencer
    rpc HandleScalogSendLocalCutResponse(ScalogSendLocalCutRequest) returns (ScalogSendLocalCutResponse);
    // Receives the global cut from global sequencer
    rpc HandleScalogSendGlobalCutResponse(ScalogSendGlobalCutRequest) returns (ScalogSendGlobalCutResponse);
}

// Request containing topic and url of the global sequencer
message ScalogStartLocalSequencerRequest {
    string topic = 1;
    string global_sequencer_url = 2;
}

// Empty response for now
message ScalogStartLocalSequencerResponse {}

// Request containing the local cut and the epoch
message ScalogSendLocalCutRequest {
    int epoch = 1;
    int local_cut = 2;
    string topic = 3;
}

// Empty response for now
message ScalogSendLocalCutResponse {}

// Request containing the global cut and topic
message ScalogSendGlobalCutRequest {
    repeated int global_cut = 1;
    string topic = 2;
}

// Empty response for now
message ScalogSendGlobalCutResponse {}